src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -366,25 +366,54 @@ private List<E> queryNextBlock() {
             if (lastValue == null) {
                 return effectiveQuery.queryList();
             }
-            CompoundValue leftHandSide = new CompoundValue();
-            CompoundValue rightHandSide = new CompoundValue();
-
+            SQLConstraint sortingFilterConstraint = null;
+            Map<Mapping, Object> previousSortingColumns = new HashMap<>();
             for (Tuple<Mapping, Boolean> sorting : effectiveQuery.orderBys) {
                 Mapping sortColumn = sorting.getFirst();
                 boolean sortAscending = sorting.getSecond().booleanValue();
                 Object value = getPropertyValue(sortColumn, lastValue);
-                if (sortAscending) {
-                    // the order by is ascending -> COLUMN > lastvalue.column
-                    leftHandSide.addComponent(sortColumn);
-                    rightHandSide.addComponent(value);
-                } else {
-                    // the order by is descending -> lastvalue.column > COLUMN
-                    leftHandSide.addComponent(value);
-                    rightHandSide.addComponent(sortColumn);
-                }
+
+                SQLConstraint currentColumConstraint = createSqlConstraintForSortingColumn(sortAscending, sortColumn,
+                        value, previousSortingColumns);
+                sortingFilterConstraint = OMA.FILTERS.or(sortingFilterConstraint, currentColumConstraint);
+
+                previousSortingColumns.put(sortColumn, value);
             }
+            return effectiveQuery.where(sortingFilterConstraint).queryList();
+        }
 
-            return effectiveQuery.where(OMA.FILTERS.gt(leftHandSide, rightHandSide)).queryList();
+        /**
+         * Creates a sql constraint for sorting purposes.
+         * </p>
+         * In MySQL/MariaDB, NULL is considered as a 'missing, unkonwn value'. Any arithmetic comparison with NULL
+         * returns false e.g. NULL != 'any' returns false.
+         * Therefore, comparisons with NULL values must be treated specially.
+         *
+         * @param sortAscending decides whether the sorting direction is descending or ascending
+         * @param column the column to be used for sorting
+         * @param value the value of the column
+         * @param previousSortingColumns all columns that should be sorted before the current one
+         * @return {@link SQLConstraint} which can be used to map a level of sorting.
+         */
+        private SQLConstraint createSqlConstraintForSortingColumn(boolean sortAscending, Mapping column, Object value,
```

Comment:
can we provide a unit test / integration test (whatever works) for this whole thing?